File: client/nodejs/src/pdu/impl/AuthenticateRequest.js
/**
 * Copyright 2016 TIBCO Software Inc. All rights reserved.
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); You may not use this file except 
 * in compliance with the License.
 * A copy of the License is included in the distribution package with this file.
 * You also may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

var util         = require('util'),
    VerbId       = require('./VerbId').VerbId,
    Request      = require('./Request').Request,
    TGException  = require('../../exception/TGException').TGException,
    TGLogManager = require('../../log/TGLogManager'),
    TGLogLevel   = require('../../log/TGLogger').TGLogLevel;

var logger = TGLogManager.getLogger();

function AuthenticateRequest() {
	AuthenticateRequest.super_.call(this);
    this._clientId  = null;
    this._inboxAddr = null;
    this._userName  = null;
    this._password  = null;
}

util.inherits(AuthenticateRequest, Request);

AuthenticateRequest.prototype.getVerbId = function() {
    return VerbId.AUTHENTICATE_REQUEST;
};

/**
 * Is this message updateable.
 */
AuthenticateRequest.prototype.isUpdateable = function() {
    return false;
};

AuthenticateRequest.prototype.writePayload = function(outputStream) {
	logger.logDebugWire(
			'AuthenticateRequest.prototype.writePayload  in');
    if (this._clientId === null || this._clientId.length === 0) {
        outputStream.writeBoolean(true);
    } else {
        outputStream.writeBoolean(false);
        outputStream.writeUTF(this._clientId);
    }
    if (this._inboxAddr === null || this._inboxAddr.length === 0) {
        outputStream.writeBoolean(true);
    }
    else {
        outputStream.writeBoolean(false);
        outputStream.writeUTF(this._inboxAddr);
    }
    if (this._username === null || this._username.length === 0) {
        outputStream.writeBoolean(true);
    }
    else {
        outputStream.writeBoolean(false);
        outputStream.writeUTF(this._username);
    }
    outputStream.writeBytes(this._password);
    logger.logDebugWire(
    		'AuthenticateRequest.prototype.writePayload  out');
};

AuthenticateRequest.prototype.readPayload = function(inputstream) {

};

AuthenticateRequest.prototype.getClientId = function() {
    return this._clientId;
};

AuthenticateRequest.prototype.setClientId = function(clientId) {
    this._clientId = clientId;
};

AuthenticateRequest.prototype.getUsername = function() {
    return this._username;
};

AuthenticateRequest.prototype.setUsername = function(username) {
    this._username = username;
};

AuthenticateRequest.prototype.getInboxAddr = function() {
    return this._inboxAddr;
};

AuthenticateRequest.prototype.setInboxAddr = function(inboxAddr) {
    this._inboxAddr = inboxAddr;
};

AuthenticateRequest.prototype.getPassword = function() {
    return this._password;
};

AuthenticateRequest.prototype.setPassword = function(password) {
    this._password = password;
};

exports.AuthenticateRequest = AuthenticateRequest;

File: api/java/src/com/tibco/tgdb/pdu/impl/AuthenticateRequest.java
/**
 * Copyright 2019 TIBCO Software Inc. All rights reserved.
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); You may not use this file except 
 * in compliance with the License.
 * A copy of the License is included in the distribution package with this file.
 * You also may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * <p/>
 * File name :AuthenticateRequest
 * Created on: 12/24/14
 * Created by: suresh
 * <p/>
 * SVN Id: $Id: AuthenticateRequest.java 4062 2020-06-04 17:48:37Z sbangar $
 */

package com.tibco.tgdb.pdu.impl;

import com.tibco.tgdb.exception.TGException;
import com.tibco.tgdb.pdu.TGInputStream;
import com.tibco.tgdb.pdu.TGOutputStream;
import com.tibco.tgdb.pdu.VerbId;
import com.tibco.tgdb.utils.TGConstants;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class AuthenticateRequest extends AbstractProtocolMessage {


    String clientId;
    String inboxAddr;
    String userName;
    String dbName;
    List<String> roles = null;
    byte[] password = TGConstants.EmptyByteArray;

    @Override
    public VerbId getVerbId() {
        return VerbId.AuthenticateRequest;
    }

    @Override
    public boolean isUpdateable() {
        return false;
    }

    @Override
    protected void writePayload(TGOutputStream os) throws TGException, IOException {
        if ((dbName == null) || dbName.length() == 0) {
            os.writeBoolean(true);
        }
        else {
            os.writeBoolean(false); //No clientId
            os.writeUTF(dbName);
        }
        if ((clientId == null) || clientId.length() == 0) {
            os.writeBoolean(true);
        }
        else {
            os.writeBoolean(false); //No clientId
            os.writeUTF(clientId);
        }
        if ((inboxAddr == null) || inboxAddr.length() == 0) {
            os.writeBoolean(true);
        }
        else {
            os.writeBoolean(false);
            os.writeUTF(inboxAddr);
        }

        if(roles == null)
             os.writeInt(-1);//use all roles
        else{
            os.writeInt(roles.size());
            for(String roleName : roles)
                os.writeUTF(roleName);
        }

        if ((userName == null) || userName.length() == 0) {
            os.writeBoolean(true);
        }
        else {
            os.writeBoolean(false);
            os.writeUTF(userName);  //Can't be null.
        }
        os.writeBytes(password);
    }

    @Override
    protected void readPayload(TGInputStream is) throws TGException, IOException {
        //For Testing purpose only.
        boolean bIsClientId = is.readBoolean();
        if (!bIsClientId) {
            this.clientId = is.readUTF();
        }
        this.inboxAddr = is.readUTF();
        this.userName = is.readUTF();
        this.password = is.readBytes();
    }


    public String getClientId() { return clientId;}
    public void setClientId(String s) { this.clientId = s;}

    public String getInboxAddr() { return inboxAddr;}
    public void setInboxAddr(String s) { this.inboxAddr = s;}

    public String getUserName() { return userName;}
    public void setUserName(String s) { this.userName = s;}

    public byte[] getPassword() { return password;}
    public void setPassword(byte[] b) { this.password = b;}

    public void setDatabaseName(String dbName) { this.dbName = dbName;}
    public String getDatabaseName() { return this.dbName; }

    public void setRoles(List<String> roleList) { this.roles = roleList; }
    public List<String> getRoles() { return this.roles; }

}